Sources/SwiftGodot/EntryPoint.swift

```diff
@@ -7,62 +7,6 @@
 
 @_implementationOnly import GDExtension
 
-public protocol ExtensionInterface {
```

Comment:
I moved this stuff to its own file for clarity.
I think the stuff in here and the new file could do with cleaning up & refactoring, but that can wait for another PR...